cli/internal/cache/cache.go

```diff
@@ -53,39 +55,42 @@ type cacheMultiplexer struct {
 }
 
 func (mplex cacheMultiplexer) Put(target string, key string, duration int, files []string) error {
-	mplex.storeUntil(target, key, duration, files, len(mplex.caches))
-	return nil
+	return mplex.storeUntil(target, key, duration, files, len(mplex.caches))
 }
 
 // storeUntil stores artifacts into higher priority caches than the given one.
 // Used after artifact retrieval to ensure we have them in eg. the directory cache after
 // downloading from the RPC cache.
 // This is a little inefficient since we could write the file to plz-out then copy it to the dir cache,
 // but it's hard to fix that without breaking the cache abstraction.
-func (mplex cacheMultiplexer) storeUntil(target string, key string, duration int, outputGlobs []string, stopAt int) {
+func (mplex cacheMultiplexer) storeUntil(target string, key string, duration int, outputGlobs []string, stopAt int) error {
 	// Attempt to store on all caches simultaneously.
-	var wg sync.WaitGroup
+	g := new(errgroup.Group)
 	for i, cache := range mplex.caches {
 		if i == stopAt {
 			break
 		}
-		wg.Add(1)
-		go func(cache Cache) {
-			cache.Put(target, key, duration, outputGlobs)
-			wg.Done()
-		}(cache)
+		c := cache
+		g.Go(func() error {
+			return c.Put(target, key, duration, outputGlobs)
+		})
+	}
+
+	if err := g.Wait(); err != nil {
+		return err
 	}
-	wg.Wait()
+
+	return nil
 }
 
 func (mplex cacheMultiplexer) Fetch(target string, key string, files []string) (bool, []string, error) {
 	// Retrieve from caches sequentially; if we did them simultaneously we could
 	// easily write the same file from two goroutines at once.
 	for i, cache := range mplex.caches {
-		if ok, actualFiles, _ := cache.Fetch(target, key, files); ok {
+		if ok, actualFiles, err := cache.Fetch(target, key, files); ok {
 			// Store this into other caches
 			mplex.storeUntil(target, key, 0, actualFiles, i)
-			return ok, actualFiles, nil
+			return ok, actualFiles, err
```

Comment:
The error we're returning here is from `cache.Fetch`, but we have just called a potentially-failing `mplex.storeUntil`. I think at the very least we ought to have a comment as to why we're ignoring that error.